include/zephyr/ipc/pbuf.h

```diff
@@ -201,6 +207,14 @@ int pbuf_write(struct pbuf *pb, const char *buf, uint16_t len);
  */
 int pbuf_read(struct pbuf *pb, char *buf, uint16_t len);
 
+#if defined(CONFIG_ARCH_POSIX)
+/*
+ * Remap the pbuf memory pointers into something which can be used in the native simulated
+ * targets
+ */
+void pbuf_native_addr_remap(struct pbuf *pb);
```

Comment:
Maybe make this function more consistent, because right now calling it depends on few conditions. It is automatically called in `init` function, but if you not calling `init` then you have to call it directly. If `CONFIG_ARCH_POSIX` is enabled, you have to call it. Otherwise, you don't.

I suggest calling this function always before using `pb`. Even before `pbuf_init`. if `CONFIG_ARCH_POSIX` it can be empty static inline function. This will make it clearer when to use it and we avoid additional `#if`s when calling it.

Reply:
Also, please update the comment to be doxygen-compatible and add statement when it must be called.

Reply:
Thank you very much for the review. I updated the PR content with what we just discussed on the phone.